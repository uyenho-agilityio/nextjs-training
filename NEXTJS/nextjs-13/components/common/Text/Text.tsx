// Libs
import { Text as TextChakra, TextProps as TextPropsChakra } from '@chakra-ui/react';
import React, { ReactNode } from 'react';

export interface TextProps extends TextPropsChakra {
  size?: string;
  variant?: string;
  children: ReactNode;
}

export const Text: React.FC<TextProps> = ({ size, variant, children, ...props }): JSX.Element => {
  return (
    <TextChakra size={size} variant={variant} {...props}>
      {children}
    </TextChakra>
  );
};
