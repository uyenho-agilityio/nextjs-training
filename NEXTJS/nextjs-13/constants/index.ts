export const LOGO = {
  alt: 'Logo',
  src: '/images/logo.svg',
  width: 32,
  height: 32,
};

export const PAGE_404 = {
  src: '/images/404.jpg',
  alt: '404 Page',
  width: 400,
  height: 400,
};

export const PAGE_ROUTES = {
  HOME: '/',
  ABOUT: '/about',
  EVENTS: '/events',
};

export const ROUTE_LIST = [
  {
    label: 'Home',
    path: '/',
  },
  {
    label: 'Events',
    path: '/events',
  },
  {
    label: 'About',
    path: '/about',
  },
];

export const MESSAGES = {
  NETWORK_ERROR: 'Network error. Please try again later.',
  ERROR: 'Something went wrong!',
};

export const TEXT = {
  CONTINUE: 'Continue',
  BACK_HOME: 'BACK TO HOME',
  HEAD_BACK: "Let's Head Back!",
  SIGN_IN: 'Sign in',
  ACADEMY: 'Academy',
};

export const SEO_DATA = {
  title: 'NextJS Training with app/ directory',
  description: 'Do examples to understand NextJS 13',
  icon: '/favicon.ico',
  url: 'https://nextjs-examples-training.vercel.app/',
  thumbnailUrl:
    'https://upload.wikimedia.org/wikipedia/commons/thumb/8/8e/Nextjs-logo.svg/1200px-Nextjs-logo.svg.png',
};

export const BASE_URL = process.env.NEXT_PUBLIC_BASE_URL;
export const LOCAL_BASE_URL = process.env.NEXT_PUBLIC_LOCAL_BASE_URL;
