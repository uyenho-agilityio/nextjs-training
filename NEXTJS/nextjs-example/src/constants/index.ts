export const VERCEL_LOGO = {
  alt: 'Vercel Logo',
  src: '/images/vercel.svg',
  width: 72,
  height: 16,
};

export const COURSE_LOGO = {
  alt: 'Logo',
  src: '/images/logo.svg',
  width: 32,
  height: 32,
};

export const AVATAR = {
  alt: 'Avatar',
  src: '/images/avatar.jpg',
  width: 40,
  height: 40,
};

export const PAGE_404 = {
  alt: '404 Page',
  src: '/images/404.jpg',
  width: 400,
  height: 400,
};

export const PAGE_ROUTES = {
  HOME: '/',
  ABOUT: '/about',
  EVENTS: '/events',
};

export const MESSAGES = {
  NETWORK_ERROR: 'Network error. Please try again later.',
  ERROR: 'Something went wrong!',
};

export const TEXT = {
  CONTINUE: 'Continue',
  BACK_TO_COURSE: 'Back to Course',
  BACK_HOME: 'BACK TO HOME',
  HEAD_BACK: "Let's Head Back!",
  SIGN_IN: 'Sign in',
  ACADEMY: 'Academy',
};
