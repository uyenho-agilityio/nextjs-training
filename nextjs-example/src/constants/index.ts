export const VERCEL_LOGO = {
  alt: 'Vercel Logo',
  src: '/images/vercel.svg',
  width: 72,
  height: 16,
};

export const COURSE_LOGO = {
  alt: 'Logo',
  src: '/images/logo.svg',
  width: 32,
  height: 32,
};

export const AVATAR = {
  alt: 'Avatar',
  src: '/images/avatar.jpg',
  width: 40,
  height: 40,
};

export const PAGE_404 = {
  alt: '404 Page',
  src: '/images/404.jpg',
  width: 400,
  height: 400,
};

export const PAGE_ROUTES = {
  HOME: '/',
  ABOUT: '/about',
  EVENTS: '/events',
};

export const ROUTE_LIST = [
  {
    label: 'Home',
    path: '/',
  },
  {
    label: 'Events',
    path: '/events',
  },
  {
    label: 'About',
    path: '/about',
  },
  {
    label: 'NextJS',
    path: '/nextjs',
  },
];

export const MESSAGES = {
  NETWORK_ERROR: 'Network error. Please try again later.',
  ERROR: 'Something went wrong!',
};

export const TEXT = {
  CONTINUE: 'Continue',
  BACK_TO_COURSE: 'Back to Course',
  BACK_HOME: 'BACK TO HOME',
  HEAD_BACK: "Let's Head Back!",
  SIGN_IN: 'Sign in',
  ACADEMY: 'Academy',
};

export const SEO_DATA = {
  title: 'NextJS Training',
  description: 'Do examples to understand NextJS',
  icon: '/images/favicon.ico',
  url: 'https://nextjs-examples-training.vercel.app/',
  thumbnailUrl:
    'https://upload.wikimedia.org/wikipedia/commons/thumb/8/8e/Nextjs-logo.svg/1200px-Nextjs-logo.svg.png',
};

export const BASE_URL = process.env.NEXT_PUBLIC_BASE_URL;
