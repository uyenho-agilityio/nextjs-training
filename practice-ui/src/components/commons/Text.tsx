import React, { ReactNode } from 'react';
import { TextVariants } from '~enums/variant';

export interface TextProps {
  className?: string;
  fs?: string;
  lh?: string;
  fontFamily?: string;
  children: ReactNode;
  variant?: TextVariants;
}

export const Text: React.FC<TextProps> = React.memo(
  ({ className, fs, lh, children, variant = TextVariants.Normal }): JSX.Element => (
    <p
      className={`${variant === TextVariants.Highlight ? `fw-bold` : ''} ${className}`}
      style={{
        fontSize: fs,
        lineHeight: lh,
      }}
    >
      {children}
    </p>
  ),
);
