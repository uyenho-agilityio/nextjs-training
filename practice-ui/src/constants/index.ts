export * from './footer';
export * from './header';
export * from './management';
export * from './pricing';
export * from './product';
