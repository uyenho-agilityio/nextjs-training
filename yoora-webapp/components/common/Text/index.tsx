'use client';

// Lib
import { Text as ChakraText, TextProps as ChakraTextProps } from '@chakra-ui/react';

type TextProps = {
  size?: string;
  variant?: string;
} & ChakraTextProps;

export const Text = ({ size, variant, ...props }: TextProps): JSX.Element => (
  <ChakraText size={size} variant={variant} {...props} />
);
