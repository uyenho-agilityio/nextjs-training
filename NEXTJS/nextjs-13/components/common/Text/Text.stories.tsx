// Lib
import Storybook from '@storybook/react';

// Component
import { Text } from './Text';

// Constant
import { TEXT } from '@webapp/constants';

export default {
  title: 'Components/Text',
  component: Text,
  argTypes: {
    backgroundColor: { control: 'color' },
  },
} as Storybook.ComponentMeta<typeof Text>;

const Template: Storybook.ComponentStory<typeof Text> = args => <Text {...args} />;

export const Paragraph = Template.bind({});
Paragraph.args = {
  children: TEXT.SIGN_IN,
};
