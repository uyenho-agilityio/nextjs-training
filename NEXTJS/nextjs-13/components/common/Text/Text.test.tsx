// Lib
import { render, screen } from '@testing-library/react';

// Component
import { Text } from './Text';

// Constant
import { TEXT } from 'constants/index';

describe('Text render', () => {
  test('should render text component with default props', () => {
    const { container } = render(<Text>{TEXT.SIGN_IN}</Text>);
    expect(container).toMatchSnapshot();
  });

  test('should render text component with valid props', () => {
    render(<Text>{TEXT.SIGN_IN}</Text>);
    expect(screen.getByText(/sign/i)).toHaveTextContent('Sign in');
  });
});
