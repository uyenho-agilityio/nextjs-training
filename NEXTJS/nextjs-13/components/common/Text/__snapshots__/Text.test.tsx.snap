// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text render should render text component with default props 1`] = `
<div>
  <p
    class="chakra-text css-0"
  >
    Sign in
  </p>
</div>
`;
